docs/rfc/007-endpoint-errors.md

```diff
@@ -0,0 +1,137 @@
+# [RFC] Conjure Endpoint Errors
+
+**Date:** 2020/03/19
+
+**Stakeholders:**
+* *Anyone that produces or consumes conjure-generated code via Dialogue or Undertow.*
+
+* * *
+
+## **Context**
+
+Currently, errors can be defined in Conjure type definitions. These errors can then be thrown by services as `ServiceException`s (which are `RuntimeException`s), and if a client knows about them, they can be caught during runtime by clients (packaged up as `RemoteException`s).
```

Comment:
Let's refactor this a bit to discuss the protocol, with a separate section describing how it impacts existing implementations (java).